openfeature-provider/java/src/main/java/com/spotify/confidence/OpenFeatureLocalResolveProvider.java

```diff
@@ -75,6 +75,53 @@ public OpenFeatureLocalResolveProvider(ApiSecret apiSecret, String clientSecret)
     this(apiSecret, clientSecret, new RemoteResolverFallback());
   }
 
+  /**
+   * Creates a new OpenFeature provider for local flag resolution with custom channel factory.
+   *
+   * <p>This constructor accepts a {@link LocalProviderConfig} which allows you to customize
+   * how gRPC channels are created, particularly useful for testing with mock servers or
+   * advanced production scenarios requiring custom connection logic.
+   *
+   * <p><strong>Example with custom channel factory for testing:</strong>
+   * <pre>{@code
+   * ChannelFactory mockFactory = (target, interceptors) ->
+   *     InProcessChannelBuilder.forName("test-server")
+   *         .usePlaintext()
+   *         .intercept(interceptors.toArray(new ClientInterceptor[0]))
+   *         .build();
+   *
+   * ApiSecret apiSecret = new ApiSecret("client-id", "client-secret");
+   * LocalProviderConfig config = new LocalProviderConfig(apiSecret, mockFactory);
+   * OpenFeatureLocalResolveProvider provider =
+   *     new OpenFeatureLocalResolveProvider(config, "client-secret");
+   * }</pre>
+   *
+   * @param config the provider configuration including API credentials and optional channel factory
+   * @param clientSecret the client secret for your application, used for flag resolution
+   *     authentication
+   */
+  public OpenFeatureLocalResolveProvider(LocalProviderConfig config, String clientSecret) {
```

Comment:
Why the client secret is not part of the local config? I suppose the whole credentials system will be changed soon so we don't need to overthink this API now